frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java

```diff
@@ -147,18 +147,41 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                     .orElseGet(() -> generateErrorResponse(emailUserIsSignedInWith, auditContext));
         } catch (AccountLockedException e) {
 
-            auditService.submitAuditEvent(
-                    FrontendAuditableEvent.AUTH_ACCOUNT_TEMPORARILY_LOCKED,
-                    auditContext,
-                    e.getErrorResponse() == ErrorResponse.ERROR_1045
-                            ? AuditService.MetadataPair.pair(
-                                    "number_of_attempts_user_allowed_to_login",
-                                    configurationService.getMaxPasswordRetries())
-                            : AuditService.MetadataPair.pair(
+            switch (e.getErrorResponse()) {
+                case ERROR_1057:
+                    int count = 0;
+                    var userProfile = userContext.getUserProfile();
+                    if (configurationService.isAuthenticationAttemptsServiceEnabled()
+                            && userProfile.isPresent()) {
+                        count =
+                                authenticationAttemptsService.getCount(
+                                        String.valueOf(userProfile.get()),
+                                        JourneyType.REAUTHENTICATION,
+                                        CountType.ENTER_EMAIL);
+                    } else {
+                        count = codeStorageService.getIncorrectEmailCount(emailUserIsSignedInWith);
+                    }
+
+                    auditService.submitAuditEvent(
+                            FrontendAuditableEvent.AUTH_REAUTH_INCORRECT_EMAIL_LIMIT_BREACHED,
+                            auditContext,
+                            AuditService.MetadataPair.pair("attemptNoFailedAt", count));
+
```

Comment:
So a question here - will the count retrieved here actually be the attempt no that someone failed at? I guess I'm thinking that when we block people, we generally delete the counts, so are we sure this won't be 0?

Reply:
(Alternative approach - populate this based on the max retries from config)